src/atsamd/serial.c

```diff
@@ -10,62 +10,97 @@
 #include "internal.h" // enable_pclock
 #include "sched.h" // DECL_INIT
 
+#if CONFIG_MACH_SAMC21
+
+#define SERCOMx SERCOM2
+#define SERCOMx_IRQn SERCOM2_IRQn
+#define SERCOMx_GCLK_ID_CORE SERCOM2_GCLK_ID_CORE
+#define ID_SERCOMx ID_SERCOM2
+
+//DECL_CONSTANT_STR("RESERVE_PINS_serial", "PA12,PA9");
```

Comment:
This adds a line of dead code (and there are a couple of other lines of dead code added later).

More generally, though, I got the impression from off-line conversations that serial was not tested on samc21.  If so, I'm not sure it is a good idea to modify serial.c, if it has not been tested.  Might be better to disable CONFIG_SERIAL in Kconfig for samc21.

Reply:
Agree, will disable for this chip.